src/libostree/ostree-repo-pull.c

```diff
@@ -4541,6 +4541,43 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
   return ret;
 }
 
+gboolean
+ostree_repo_pull_with_options (OstreeRepo             *self,
+                               const char             *remote_name_or_baseurl,
+                               GVariant               *options,
+                               OstreeAsyncProgress    *progress,
+                               GCancellable           *cancellable,
+                               GError                **error)
+{
+  g_autoptr(GError) local_error = NULL;
+
+  if (!ostree_repo_pull_with_options_internal (self, remote_name_or_baseurl,
+                                               options, progress, cancellable,
+                                               &local_error))
+    {
+      if (g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND) &&
+          (options == NULL ||
+           g_variant_lookup_value (options, "disable-static-deltas", NULL) == NULL))
```

Comment:
What if you passed `disable-static-deltas` as `FALSE`? Seems like you should still get to use this fallback. Also, what about `require-static-deltas`? You shouldn't use the fallback if that's `TRUE`.

Reply:
Good catches.

Reply:
Fix force-pushed.